src/utils/metric/metric-factory.ts

```diff
@@ -50,15 +50,20 @@ export class MetricFactory {
 	 * Use a total counter when a value represents the total value of a count.
 	 * Returns null if key normalization fails to produce a valid metric key.
 	 */
-	public createTotalCounter(name: string, dimensions: Dimension[], value: number, ts?: number): Metric | null {
+	public createTotalCounter(
+		name: string,
+		dimensions: Dimension[],
+		value: number,
+		timestamp?: number
```

Comment:
Please also mention this in the comment and add "if no explicit timestamp is provided, the server will assume the current time at ingestion" or something like that.

Also update the other occurrences of `ts` below to be consistent including the respective comment.